packages/lib/src/services/bbmriNegotiate.ts

```diff
@@ -1,164 +1,127 @@
-     */
-    const returnURL: string = `${window.location.protocol}//${window.location.host}`;
+export async function bbmriNegotiate(
+    sitesToNegotiate: string[],
+): Promise<void> {
+    const currentLensOptions = get(lensOptions);
+    if (currentLensOptions.negotiateOptions === undefined) {
+        console.error('"negotiateOptions" is missing the lens options');
+        errorChannel.set('"negotiateOptions" fehlt in den Lens-Optionen');
+        return;
+    }
 
-    let response!: Response;
+    const bbmriCollectionResources: BbmriCollectionResource[] = [];
+    for (const siteMapping of currentLensOptions.negotiateOptions
+        .siteMappings) {
+        if (sitesToNegotiate.includes(siteMapping.site)) {
+            bbmriCollectionResources.push({
+                id: siteMapping.collection,
+                name: siteMapping.collection,
+                organization: {
+                    id: 0,
+                    externalId: siteMapping.site_id,
+                    name: siteMapping.site,
+                },
+            });
+        }
+    }
 
-    const BBMRI_collection_resource: BbmriCollectionResource[] = [];
+    const bbmriNegotiatorRequest: BbmriNegotiateRequest = {
+        humanReadable: getHumanReadableQuery(),
+        url: `${window.location.protocol}//${window.location.host}`,
+        resources: bbmriCollectionResources,
+    };
 
-    collections.forEach(function (collection) {
-        BBMRI_collection_resource.push({
-            id: collection.collection,
-            name: collection.collection,
-            organization: {
-                id: 0,
-                externalId: collection.site_id,
-                name: collection.site,
-            },
-        });
-    });
+    await sendRequestToNegotiator(
+        currentLensOptions.negotiateOptions.url,
+        bbmriNegotiatorRequest,
+    );
+}
 
+/**
+ * Sends the request to the BBMRI Negotiator to start the application process.
+ * In case of success the user is redirected to the BBMRI Negotiator to continue
+ * the application process. In case of a network error or if the BBMRI
+ * Negotiator responds with an unexpected HTTP status code an error is shown to
+ * the user.
+ * @param url The request endpoint for starting the application process
+ * @param bbmriNegotiateRequest The request payload
+ */
+async function sendRequestToNegotiator(
+    url: string,
+    bbmriNegotiateRequest: BbmriNegotiateRequest,
+): Promise<void> {
+    let response;
     try {
-        response = await fetch(`${negotiateOptions.url}`, {
+        // Swagger documentation: https://negotiator.bbmri-eric.eu/api/swagger-ui/index.html#/Requests/add
+        response = await fetch(url, {
             method: "POST",
             headers: {
                 Accept: "application/json; charset=utf-8",
                 "Content-Type": "application/json",
                 Authorization:
                     "Basic YmJtcmktZGlyZWN0b3J5Omw5RFJLVUROcTBTbDAySXhaUGQ2",
```

Comment:
While on a cleanup, this should also be set via the options. I know that it will be removed at some time.

Reply:
What do you mean? The swagger URL? The `negotiateOptions.url` is still configurable in the lens options, it's just passed to that function as `url`.

Reply:
I pined the comment on line 149/107, there the basic auth password is shown. In the future I would prefer to set it to empty instead of setting it directly in the code